### ISSUE TITLE
Tab customization and retab

### ISSUE THREAD

**Issue by kuizeo** [helix-editor/helix#2576]
#### Describe your feature request

In Vim (and derivatives), the `shiftwidth`, `tabstop`, and `expandtab` options to customize the user's experience with using TAB in files. In addition, `:retab` (most often `:%retab`) is used to make tabs consistent throughout the file, acting like a "stripped-down" version of Helix's `:format`. An equivalent to both (other than `s`) should exist in Helix.

**Comment by frexsdev** [comment]
:indent-style

**Comment by kuizeo** [comment]
this isn't enough - one should be able to set what a tab character represents in the file (e.x. 1 tab) **and** how many spaces that displays as (e.x. 2 spaces).

**Comment by giannitedesco** [comment]
This is covered in #1291 but basically you can set it in `languages.toml` per [the manual](https://docs.helix-editor.com/languages.html#language-configuration)

**Comment by ThomasBrierley** [comment]
> This is covered in #1291 but basically you can set it in `languages.toml` per [the manual](https://docs.helix-editor.com/languages.html#language-configuration)

I think that affects the actual buffer rather than the rendered text output?

e.g I don't want to change the actual text of a file indented with tab characters, but I would like to configure the _rendered_ width of those tab characters to x spaces etc. Everyone has a different preference for rendered tab width which is why it needs be configurable.

**Comment by hsn10** [comment]
I don't believe that configuring language settings is right answer to this problem.  It will need autoformat on save on to work + correct settings for an extension.

Expected workflow is: load file with tabs: set tab width from ':' line, reformat file and save.

I can't find a way how to trigger auto-format manually and setting indent using ':' command is not very user friendly. why bother with unit, can't we just set tab width using number, reformat and save?

**Comment by dddraxxx** [comment]
Same. I really like the setting of helix but I think it needs to support more options

**Comment by dansalvato** [comment]
This was causing issues for me as well. Pressing Tab in insert mode always inserts the same number of spaces, when it should instead insert however many spaces are needed to reach the next tabstop. Of course, I can set the indent unit to `\t` if I want this, but that also overrides the LSP format configuration, making it impossible to convert tabs to spaces on save.

As it is, there are certain file formats I can't use Helix to edit because they use spaces for indentation, but also expect tabstops to be respected when indenting.

**Comment by athre0z** [comment]
There really needs to be a way to customize the number of spaces rendered per tab in an ad-hoc / per-session kind of way. Using helix to quickly peek into source files with exotic intend configs (e.g. GNU sources with 8 spaces per tab but space padding for smaller indents) is currently a terrible experience.

**Comment by kkonevets** [comment]
common, we need spaces everywhere https://youtu.be/SsoOG6ZeyUI?t=34

**Comment by carlosvigil** [comment]
If a default for unspecified languages should be chosen, tabs are the more accessible. People who use accessibility tools have serious difficulties when spaces are forced.

**Comment by Rudxain** [comment]
#15391

**Comment by diktomat** [comment]
In addition, the current handling forces one to set tab-width for every language separately. The beauty of indention using tabs instead of spaces is the ability to choose my preferred tab-width without it affecting other developers of that code, so I should be able to set a global width for everything `\t`, switching to different tab-widths in `languages.toml` (or the proposed `:set tw`) only occasionally.

The current state is a dozen entries just of this kind in my `languages.toml`, which is kinda numb:
```toml
[[language]]
name = "bash"
indent = { tab-width = 4, unit = "\t" }
```

**Comment by hsn10** [comment]
why not be vim compatible:

```vim
set tabstop
set (no)expandtab
%retab!
```

this is proposed to be in addition to an existing [[language]] config. 

In standard developer workflow it would be better for real world applications to have tab config per project directory/language pair.

We are using different setups:

1. our current modern rust code uses tabstop=3 expandtab
2. opensource rust libraries we are contributing to are using tabstop=2 expandtab
3. our old java1.1-1.3 code is using tabstop=8 noexpandtab

For me this issue is high priority because I am always using retab! before commit.
